include/SSVOpenHexagon/Utils/Utils.hpp

```diff
@@ -83,6 +84,13 @@ inline void runLuaFile(Lua::LuaContext& mLua, const std::string& mFileName)
     }
 }
 
+struct Nothing
+{
+};
```

Comment:
What's the purpose of the Nothing struct?

Reply:
It's a workaround for C++ silliness, as `void` is not a regular type and you cannot create instances of it. `runLuaFunctionIfExists` is supposed to return a `std::optional<T>` where `T` is the return value of the Lua function. This is great if the Lua function actually returns something, but if it returns `void`, the type `std::optional<void>` is invalid. Therefore I perform a transformation from `void` to `Nothing` during the function, so that I can homogeneously handle `void` as any other type, and stick it inside a `std::optional`.